internal/handlers/common/params.go

```diff
@@ -147,6 +154,48 @@ func GetWholeNumberParam(value any) (int64, error) {
 	}
 }
 
+// GetLimitStageParam returns $limit stage argument from the provided value.
+// It returns the proper error if value doesn't meet requirements.
+func GetLimitStageParam(value any) (int64, error) {
+	max, err := GetWholeNumberParam(value)
+
+	switch {
+	case err == nil:
+	case errors.Is(err, errUnexpectedType):
+		return 0, commonerrors.NewCommandErrorMsg(
+			commonerrors.ErrStageLimitInvalidArg,
+			fmt.Sprintf("invalid argument to $limit stage: Expected a number in: $limit: %#v", value),
+		)
+	case errors.Is(err, errNotWholeNumber), errors.Is(err, errInfinity):
+		return 0, commonerrors.NewCommandErrorMsg(
+			commonerrors.ErrStageLimitInvalidArg,
+			fmt.Sprintf("invalid argument to $limit stage: Expected an integer: $limit: %#v", value),
+		)
+	case errors.Is(err, errLongExceeded):
+		return 0, commonerrors.NewCommandErrorMsg(
+			commonerrors.ErrStageLimitInvalidArg,
+			fmt.Sprintf("invalid argument to $limit stage: Cannot represent as a 64-bit integer: $limit: %#v", value),
+		)
+	default:
+		panic(fmt.Sprint("Unexpected error: ", err))
+	}
+
+	switch {
+	case max < 0:
```

Comment:
```suggestion
	case limit < 0:
```
Just so it's easier to read :hugs: